lib/ipaddr.h

```diff
@@ -174,6 +174,21 @@ static inline bool ipaddr_is_same(const struct ipaddr *ip1,
 	return ipaddr_cmp(ip1, ip2) == 0;
 }
 
+static inline bool ipaddr_is_mcast(const struct ipaddr *ip)
+{
+	switch (ip->ipa_type) {
+	case IPADDR_NONE:
```

Comment:
What exactly is IPADDR_NONE and how is it being used in this context?  Doesn't this belong in it's own commit?  This first commit is a grab bag of different changes that probably needs to be broken out